### ISSUE TITLE
Conflicting declarations for "name" property in 3.0.0-beta04

### ISSUE THREAD

**Issue by sproctor** [apollographql/apollo-android#3625]
**Summary**
If you have a GraphQL type with a column named "name", Apollo generates an enum class with a property "name" that conflict with the "name" from Kotlin's enum class.

**Version**
3.0.0-beta04, didn't happen in older version

**Comment by martinbonnin** [comment]
Hi ! Thanks for reporting this ! `beta04` changed the default generation mode of enums to use Kotlin enums instead of sealed classes (https://github.com/apollographql/apollo-android/pull/3551).

You can fallback to the previous behaviour by adding this to your Gradle file:

```kotlin
apollo {
  sealedClassesForEnumsMatching.set(listOf(".*"))
}
```

That being said, it'd be interesting to dig more into why you had the issue. We certainly need to escape some enum values. We'll look into it.

**Comment by sproctor** [comment]
`name` and `ordinal` are properties of Kotlin enums and need to be renamed. https://kotlinlang.org/docs/enum-classes.html#working-with-enum-constants

That workaround works and is good enough for me, thanks.

**Comment by BoD** [comment]
A fix for this issue has just been merged - it will be available in the next version of the v3 (can also be tested with the latest Snapshot).

**Comment by sproctor** [comment]
This is fixed in 3.0.0-beta05, thanks.
